### ISSUE TITLE
Data Issue

### ISSUE THREAD

**Issue by Arthas0310** [pinouche/ubiquant_kaggle#1]
Hi Thomas - I'm a student at UC Berkeley, and was trying to learn from the Ubiquant competition. I wish to apply some insights from the competition into my own thesis research, and was wondering whether you are open to sharing the code to generate the data used in your code. Specifically, I would really appreciate if you could share the code for pca_train.p and list_models_quantiles_scaled.p under the pickle-files folder.

Thank you so much! My email: liuyihan0310@gmail.com

Best regards,
Alex

**Comment by pinouche** [comment]
Hi Alex,

Unfortunately, I cannot find the notebook on Kaggle anymore.. However I can send you the pickle files by email (be careful when strangers send you pickle files in general, as they can contain arbitrary executable..).

I can tell you a bit more on how they are created though:

1) pca_train.p is simply fitting a pca model on the whole training set. In this notebook, this pca model is then applied in the same way to the testing set.

2) list_models_quantiles_scaled.p. This is a list of lgbm models trained until different training loss values. Similarly to the pca model, they are trained on the training data pre-processed in the same way as the test data in this notebook. The custom loss used in the lgbm is the one in the notebook. 

I performed some hyperparameter search in order to tune the lgbm models.

Hope that helps!

Thomas

**Comment by Arthas0310** [comment]
Hi Thomas,

Thank you so much for helping me with my research! It really helps me to understand the pipeline. I will check the notebook and hopefully get a better understanding of your ideas.

Best regards,
Alex
